test/game_manager_test.exs

```diff
@@ -0,0 +1,149 @@
+defmodule GameManagerTest do
+  use ExUnit.Case
+  import Mock
+  import GameManager
+  import ExUnit.CaptureIO
+  doctest GameManager
+
+  @invalid_input_message "You entered an invalid move. Please try again.\n"
```

Comment:
Is there a way that we could change our messaging that wouldn't force us to assert on exact strings like this?

Reply:
I created a function in `GameIO` that prints an invalid input message, and am now testing that this function gets called by my input validator when invalid input is entered. Do you think that's a better pattern?

Reply:
...but then I'm back to testing implementation over functionality. Just was looking back over it.